### DISCUSSION TITLE
TuxClocker 1.5.0

### DISCUSSION THREAD

**Discussion by Lurkki14** [Lurkki14/tuxclocker#88] [Announcements]
# Changelog
## New features
- Added NVIDIA clock setting outside Xorg through NVML (possibly broken on some GPUs/NVML versions)
- RX 7000-series fan speed control

## Bug fixes
- Fixed incorrect power usage reading on some CPUs
- The bundled build now works with NVIDIA; see README for how to run

## Misc. changes
- Added option to disable building CPU plugin
- Added some fields to desktop file (tujhen)
- Two decimals are now shown in the GUI for readable values

## Translations
- Finnish
- French (mrintrepide)
- Georgian (NorwayFun)
- German (Foliant2, Termuellinator)

<hr /><em>This discussion was created from the release <a href='https://github.com/Lurkki14/tuxclocker/releases/tag/1.5.0'>TuxClocker 1.5.0</a>.</em>

**Comment by lafluer56** [comment]
When i try to change the fan mode to manual and save, the fan mode line goes red.
I set the fan to 100%, but nothing happens. I am doing something wrong ?
I have the kernel 6.7.1 and a 7900xt with the open source drivers.

**Reply by Lurkki14** [reply]
Can you post output of `cat /sys/class/drm/card*/device/gpu_od/fan_ctrl/fan_curve` ?

**Reply by lafluer56** [reply]
OD_FAN_CURVE:
0: 0C 0%
1: 0C 0%
2: 0C 0%
3: 0C 0%
4: 0C 0%
OD_RANGE:
FAN_CURVE(hotspot temp): 25C 100C
FAN_CURVE(fan speed): 15% 100%

**Reply by Lurkki14** [reply]
```
echo "0 50 15" | sudo tee /sys/class/drm/card*/device/gpu_od/fan_ctrl/fan_curve # first point on the curve
echo "1 50 15" | sudo tee /sys/class/drm/card*/device/gpu_od/fan_ctrl/fan_curve # second point on the curve
echo "2 50 15" | sudo tee /sys/class/drm/card*/device/gpu_od/fan_ctrl/fan_curve # third point on the curve
echo "3 50 15" | sudo tee /sys/class/drm/card*/device/gpu_od/fan_ctrl/fan_curve # fourth point on the curve
echo "4 50 15" | sudo tee /sys/class/drm/card*/device/gpu_od/fan_ctrl/fan_curve # fifth point on the curve
echo "c" | sudo tee /sys/class/drm/card*/device/gpu_od/fan_ctrl/fan_curve # finally committing the fan curve
```
Can you set in on the terminal like this? The `15` is the target speed

**Reply by Lurkki14** [reply]
I think I see the failure here, we are parsing the current fan curve and not changing the temperatures, but trying to set the temperature it shows is not actually within range.

**Reply by Lurkki14** [reply]
Also, does setting fan mode to Auto in TuxClocker after doing that reset the fan speed?

**Reply by lafluer56** [reply]
Ok, if i run the echo commands all works good.
If i set auto, manual in TuxClocker it doesn t erase the other commands.

Another problem is the display info in TuxClocker, the fan speed value seems ok, i get 100%, but for the mode, it displays 'no mode set'.

**Reply by Lurkki14** [reply]
Does `echo "r" | sudo tee /sys/class/drm/card*/device/gpu_od/fan_ctrl/fan_curve` reset the fan speed?

**Reply by lafluer56** [reply]
yes
